packages/hdwallet-provider/src/index.ts

```diff
@@ -170,7 +170,13 @@ class HDWalletProvider {
               )
             };
           }
-          const tx = Transaction.fromTxData(txParams, txOptions);
+
+          // Taken from https://github.com/ethers-io/ethers.js/blob/2a7ce0e72a1e0c9469e10392b0329e75e341cf18/packages/abstract-signer/src.ts/index.ts#L215
+          const hasEip1559 = (txParams.maxFeePerGas !== undefined || txParams.maxPriorityFeePerGas !== undefined);
```

Comment:
Maybe we should also ensure that `type: "0x02"` is in the tx params as well if we detect that it is 1559. In other words maybe we should merge that param in after we find out it is a 1559 tx.

Reply:
Because new transactions must have `maxFeePerGas` and `maxPriorityFeePerGas` parameters.  And the new transaction type will be automatically set(replaced) by `FeeMarketEIP1559Transaction`. So we don’t need to set `type: "0x02"`

https://github.com/ethereumjs/ethereumjs-monorepo/blob/ae9989c6efd5d7916938351ae0f065b9fc9adb42/packages/tx/src/eip1559Transaction.ts#L181-L182

Reply:
Ah ok cool, in the README examples they set the type manually. Looks good!